### DISCUSSION TITLE
How to let third-party enum confirm CasePathable

### DISCUSSION THREAD

**Discussion by marcuswu0814** [pointfreeco/swift-case-paths#144] [Q&A] [answered]
In latest version, the `extract(case:from:)` be [deprecated](https://swiftpackageindex.com/pointfreeco/swift-case-paths/main/documentation/casepaths/extract%28case:from:%29-89rof), if I want to extract a third-party enum, what’s is the alternative way to do? 🙏

Seems not possible to add macro with `extension` like this:

```swift
@CasePathable
extension ThirdPartyEnum {} // Error: 'extension' macro cannot be attached to extension
```

**Comment by stephencelis** [comment, marked as answer]
Hi @marcuswu0814! It's not possible to apply a macro to define such a conformance, but it is possible to define it from scratch for types you don't own. For example, the library defines `CasePathable` conformances for the `Optional` and `Result` types: https://github.com/pointfreeco/swift-case-paths/blob/main/Sources/CasePaths/Result%2BCasePathable.swift

We also define a bunch of manual conformances in the Composable Architecture.
